modules/kargoBidAdapter.js

```diff
@@ -4,286 +4,484 @@ import { registerBidder } from '../src/adapters/bidderFactory.js';
+          };
+        default:
+          suaValidAttributes.push(suaKey);
+      }
     }
 
+    krakenParams.device.sua = pick(uaClientHints, suaValidAttributes);
+  }
+
+  const validPageId = getLocalStorageSafely(CERBERUS.PAGE_VIEW_ID) != null 
+  const validPageTimestamp = getLocalStorageSafely(CERBERUS.PAGE_VIEW_TIMESTAMP) != null
+  const validPageUrl = getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL) != null
+
+
+  const page = {}
+  if(validPageId) {
+    page.id = getLocalStorageSafely(CERBERUS.PAGE_VIEW_ID)
+  }
+  if(validPageTimestamp) {
+    page.timestamp = Number(getLocalStorageSafely(CERBERUS.PAGE_VIEW_TIMESTAMP))
+  }
+  if(validPageUrl) {
+    page.url = getLocalStorageSafely(CERBERUS.PAGE_VIEW_URL)
+  }
+  if (!_.isEmpty(page)) {
+    krakenParams.page = page
+  }
+  
+  return Object.assign({}, bidderRequest, {
+    method: BIDDER.REQUEST_METHOD,
+    url: `${BIDDER.HOST}${BIDDER.REQUEST_ENDPOINT}`,
+    data: krakenParams,
+    currency: currency
+  });
+}
+
+function interpretResponse(response, bidRequest) {
+  let bids = response.body;
+  const bidResponses = [];
+
+  if (_.isEmpty(bids)) {
+    return bidResponses;
+  }
+
+  if (typeof bids !== 'object') {
     return bidResponses;
-  },
-  getUserSyncs: function(syncOptions, responses, gdprConsent, usPrivacy) {
-    const syncs = [];
-    const seed = spec._generateRandomUuid();
-    const clientId = spec._getClientId();
-    var gdpr = (gdprConsent && gdprConsent.gdprApplies) ? 1 : 0;
-    var gdprConsentString = (gdprConsent && gdprConsent.consentString) ? gdprConsent.consentString : '';
-    // don't sync if opted out via usPrivacy
-    if (typeof usPrivacy == 'string' && usPrivacy.length == 4 && usPrivacy[0] == 1 && usPrivacy[2] == 'Y') {
-      return syncs;
+  }
+
+  Object.entries(bids).forEach((entry) => {
+    const [bidID, adUnit] = entry
+
+    let meta = {
+      mediaType: adUnit.mediaType && BIDDER.SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType) ? adUnit.mediaType : BANNER,
+    };
+
+    if (adUnit.metadata && adUnit.metadata.landingPageDomain) {
+      meta.clickUrl = adUnit.metadata.landingPageDomain[0];
+      meta.advertiserDomains = adUnit.metadata.landingPageDomain;
     }
-    if (syncOptions.iframeEnabled && seed && clientId) {
-      for (let i = 0; i < SYNC_COUNT; i++) {
-        syncs.push({
-          type: 'iframe',
-          url: SYNC.replace('{UUID}', clientId).replace('{SEED}', seed)
-            .replace('{INDEX}', i)
-            .replace('{GDPR}', gdpr)
-            .replace('{GDPR_CONSENT}', gdprConsentString)
-            .replace('{US_PRIVACY}', usPrivacy || '')
-        });
-      }
+
+    const bidResponse = {
+      requestId: bidID,
+      cpm: Number(adUnit.cpm),
+      width: adUnit.width,
+      height: adUnit.height,
+      ttl: 300,
+      creativeId: adUnit.id,
+      dealId: adUnit.targetingCustom,
+      netRevenue: true,
+      currency: adUnit.currency || bidRequest.currency,
+      mediaType: meta.mediaType,
```

Comment:
Won't this be storing a boolean?

```
let meta = {
      mediaType: adUnit.mediaType && BIDDER.SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType) ? adUnit.mediaType : BANNER,
    };
```

Sorry if I'm confusing the syntax, but `adUnit.mediaType && BIDDER.SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType) ? adUnit.mediaType : BANNER` is checking if the adUnits media type is the same as the bidder supported media type and would set a boolean. 

In the current adapter code its used like this:
```
if (adUnit.mediaType && SUPPORTED_MEDIA_TYPES.includes(adUnit.mediaType)) {
        meta.mediaType = adUnit.mediaType;
}
```

Reply:
In that statement the `?` is a ternary operator so yes the part before it resolves to a boolean. After the the ternary theres two possible return values for when the boolean is true or false.

`result = (A = B) ? <RETURN X if A = B> :  <RETURN Y if A != B>`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator